go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -280,6 +280,197 @@ func (c *Controller) reconcile(name string) error {
+// across different RAs
+func (c *Controller) checkSubnetOverlaps(ra *ratypes.RouteAdvertisements, selectedNetworks *selectedNetworks) error {
+	NetworkVRFsSubnets := createNetworkVRFsSubnetsMap(ra, selectedNetworks)
+
+	for networkVRF, currentRASubnets := range NetworkVRFsSubnets {
+		// Get all other RAs and their subnets for the same VRF
+		allRAs, err := c.raLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+
+		foundOtherRA := false
+		for _, otherRA := range allRAs {
+			if otherRA.Name == ra.Name {
+				continue
+			}
+			otherSelectedNetworks, err := c.getSelectedNetworksForRA(otherRA)
+			if err != nil {
+				// If we can't get networks from another RA, log but don't fail current RA reconcile.
+				// This could happen if the other RA has configuration issues
+				klog.V(4).Infof("Failed to get selected networks from RouteAdvertisements %q: %v", otherRA.Name, err)
+				continue
+			}
+
+			otherRANetworkVRFsSubnets := createNetworkVRFsSubnetsMap(otherRA, otherSelectedNetworks)
+			otherSubnets, hasMatchingVRF := otherRANetworkVRFsSubnets[networkVRF]
+			if !hasMatchingVRF || len(otherSubnets) == 0 {
+				continue
+			}
+
+			foundOtherRA = true
+			combinedSubnets := append(currentRASubnets, otherSubnets...)
+			overlappingIndices, err := checkSubnetSliceOverlapsWithIndices(combinedSubnets)
+			if err != nil {
+				return fmt.Errorf("%w: failed to check subnet overlap: %w", errConfig, err)
+			}
+
+			if len(overlappingIndices) > 0 {
+				subnet1Index, subnet2Index := overlappingIndices[0], overlappingIndices[1]
+				subnet1, subnet2 := combinedSubnets[subnet1Index], combinedSubnets[subnet2Index]
+				currentRASubnetsLen := len(currentRASubnets)
+
+				// Check which RA each subnet belongs to based on index
+				subnet1InCurrent := subnet1Index < currentRASubnetsLen
+				subnet2InCurrent := subnet2Index < currentRASubnetsLen
+
+				if subnet1InCurrent && subnet2InCurrent {
+					// Same-RA conflict
+					return fmt.Errorf("%w, overlapping CIDR detected within RouteAdvertisement %q in VRF %q: [%s %s]", errConfig, ra.Name, networkVRF, subnet1, subnet2)
+				} else {
+					// Cross-RA conflict - ensure current RA subnet is listed first
+					var currentSubnet, otherSubnet string
+					if subnet1InCurrent {
+						currentSubnet = subnet1
+						otherSubnet = subnet2
+					} else {
+						currentSubnet = subnet2
+						otherSubnet = subnet1
+					}
+
+					conflictErr := fmt.Errorf("%w: overlapping CIDR detected between RouteAdvertisements %q and %q in VRF %q: [%s %s]", errConfig, ra.Name, otherRA.Name, networkVRF, currentSubnet, otherSubnet)
+
+					// Update status of the conflicting RA as well
+					if err := c.updateRAStatus(otherRA, false, conflictErr); err != nil {
+						klog.Errorf("Failed to update status of conflicting RouteAdvertisements %q: %v", otherRA.Name, err)
+					}
+
+					return conflictErr
+				}
+			}
+		}
+
+		// If no other RAs were found, still need to check for same-RA conflicts
```

Comment:
Just add it at the beginning unconditionally, we don't care what overlapping we detect first.

Reply:
Using the boolean would save us one loop in most cases( where there is more than 1 RA) but we could do it anyway it's not super important..